lib/src/top_level.rs

```diff
@@ -14,6 +14,10 @@ use rustc_span::symbol::sym;
 use std::collections::HashMap;
 use std::string::ToString;
 
+pub struct TopLevelOptions {
```

Comment:
```suggestion
pub(crate) struct TopLevelOptions {
```
trying to do `pub(crate)` everywhere

Reply:
This adds better detection of dead code with the Rust linter.